src/backend/distributed/operations/shard_split.c

```diff
@@ -575,6 +575,9 @@ BlockingShardSplit(SplitOperation splitOperation,
 	}
 	PG_CATCH();
 	{
+		/* end ongoing transactions to enable us to clean up */
+		ShutdownAllConnections();
```

Comment:
without this I ended up in a deadlock in failure tests, because worker_split_copy might still be in progress

we should prevent transaction blocks, because transaction semantics break down completely at this point

Reply:
Will this cancel the connections of worker_split_copy to avoid the deadlock?

Reply:
yes, shutting down all the sessions also releases the locks